### PR TITLE
New scripts and tests for GeoMechanicsApp

### PR DESCRIPTION
**Pull request by Vahid-Galavi**
**📝 Description**
Three new tests are added. In addition, a python script is added to solve staged analysis when input files are stored in seperated folders.

### CODE REVIEW

File: applications/GeoMechanicsApplication/tests/test_element_lab.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    def setUp(self):
+        # Code here will be placed BEFORE every test in this TestCase.
+        pass
+
+    def tearDown(self):
+        # Code here will be placed AFTER every test in this TestCase.
+        pass
+
+    def test_triaxial_comp_6n(self):
```
Comment (aronnoordam) [inline applications/GeoMechanicsApplication/tests/test_element_lab.py:19]: Can you describe what the test does in the docstring?

File: applications/GeoMechanicsApplication/tests/test_element_lab.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        effective_stresses_xx = [integration_point[0,0] for element in effective_stresses_stages[stage_nr] for integration_point in element]
+        effective_stresses_yy = [integration_point[1,1] for element in effective_stresses_stages[stage_nr] for integration_point in element]
+        effective_stresses_zz = [integration_point[2,2] for element in effective_stresses_stages[stage_nr] for integration_point in element]
+        # Assert integration point information
+        for idx, effective_stress_xx in enumerate(effective_stresses_xx):
+            self.assertAlmostEqual(-100.0, effective_stress_xx,        2)
+            self.assertAlmostEqual(-300.0, effective_stresses_yy[idx], 2)
+            self.assertAlmostEqual(-100.0, effective_stresses_zz[idx], 2)
+
+    def test_oedometer_ULFEM(self):
```
Comment (aronnoordam) [inline applications/GeoMechanicsApplication/tests/test_element_lab.py:62]: Can you describe what the test does in the docstring?

File: applications/GeoMechanicsApplication/tests/test_element_lab.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            self.assertAlmostEqual(0.0,      effective_stress_xx,        3)
+            self.assertAlmostEqual(-1000000, effective_stresses_yy[idx], 2)
+            self.assertAlmostEqual(0.0,      effective_stresses_zz[idx], 3)
+
+        y_displacements = [displacement[1] for displacement in displacements]
+        top_node_nbrs = [1]
+        for top_node_nbr in top_node_nbrs:
+            self.assertAlmostEqual(-0.0909090909516868, y_displacements[top_node_nbr], 6)
+
+    def test_oedometer_ULFEM_diff_order(self):
```
Comment (aronnoordam) [inline applications/GeoMechanicsApplication/tests/test_element_lab.py:87]: Can you describe what the test does in the docstring?

File: applications/GeoMechanicsApplication/tests/test_element_lab.py (hunk 1 above)
Comment (aronnoordam) [reply applications/GeoMechanicsApplication/tests/test_element_lab.py:19]: e.g. tests mohrcoulomb model on 2d/3d column/block with lineload/pointload on top/bottom. With free side boundaries.

File: applications/GeoMechanicsApplication/tests/test_inclined_beam_2D3N.gid/MainKratos.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import KratosMultiphysics
```
Comment (aronnoordam) [inline applications/GeoMechanicsApplication/tests/test_inclined_beam_2D3N.gid/MainKratos.py]: why did you add this file?

**Comment by aronnoordam** [comment]
CI / centos is failing, I don't know what this means and/or how it is caused, because non-Geomechanics applications are failing

**Comment by roigcarlo** [comment]
@aronnoordam @Vahid-Galavi The centos fail is because this branch was created before we added support for python 3.6+ in centos. Please merge with the master and it should be fixed.

File: applications/GeoMechanicsApplication/tests/test_element_lab.py (hunk 1 above)
Comment (Vahid-Galavi) [reply applications/GeoMechanicsApplication/tests/test_element_lab.py:19]: I added a description for the test

File: applications/GeoMechanicsApplication/tests/test_element_lab.py (hunk 3 above)
Comment (Vahid-Galavi) [reply applications/GeoMechanicsApplication/tests/test_element_lab.py:87]: I added a description for the test

File: applications/GeoMechanicsApplication/tests/test_inclined_beam_2D3N.gid/MainKratos.py (hunk 4 above)
Comment (Vahid-Galavi) [reply applications/GeoMechanicsApplication/tests/test_inclined_beam_2D3N.gid/MainKratos.py]: I will delete the file

**Comment by Vahid-Galavi** [comment]
I realised that there was a minor issue to apply boundary conditions for variables, not defined as DOF.

**Comment by Vahid-Galavi** [comment]
@aronnoordam I made all the requested changes. In addition, I found that there was a minor issue with imposing BC of variables, not defined as DOF.

**Comment by aronnoordam** [comment]
One of the tests in the CI ubuntu-intel job is failing in the Kratos Core. It seemed like it happened after merging the master into this branch. @roigcarlo is this expected?

**Comment by roigcarlo** [comment]
No, looking into it

**Comment by roigcarlo** [comment]
Seems to be a race condition. I am having difficulties to reproduce it.

I've rescheduled the failing job. Since seems to be very occasional chances are it just will give you OK this time. I will push the fix in another PR as it may take some time to find...

**Comment by roigcarlo** [comment]
@aronnoordam will be soved by #10565

**Review by aronnoordam — APPROVED** [review]

**Comment by Vahid-Galavi** [comment]
Thanks @aronnoordam
